ci/validate-version-bump.sh

```diff
@@ -19,43 +16,4 @@ head_sha=$(git rev-parse "${HEAD_SHA:-HEAD}")
 echo "Base branch  is $base_sha"
 echo "Current head is $head_sha"
 
-# Gets crate names of members that has been changed from $bash_sha to $head_sha.
-changed_crates=$(
-    git diff --name-only "$base_sha" "$head_sha" -- crates/ credential/ benches/ \
-    | cut  -d'/' -f2 \
-    | sort -u
-)
-
-if  [ -z "$changed_crates" ]
-then
-    echo "No file changed in member crates."
-    exit 0
-fi
-
-# Checks publish status for only crates with code changes.
-publish_status_table=$(
-    echo "$changed_crates" \
-    | xargs printf -- '--package %s\n' \
-    | xargs cargo unpublished
-)
-
-# "yes" -> code changed but no version difference -> need a bump
-# Prints 2nd column (sep by space), which is the name of the crate.
-crates_need_bump=$(
-    echo "$publish_status_table" \
-    | { grep '| yes ' || true; } \
-    | awk '{print $2}'
-)
-
-if  [ -z "$crates_need_bump" ]
-then
-    echo "No version bump needed for member crates."
-    exit 0
-fi
-
-echo "Detected changes in these crates but no version bump found:"
-echo "$crates_need_bump"
-echo
-echo "Please bump at least one patch version for each corresponding Cargo.toml:"
-echo 'Run "cargo unpublished" to read the publish status table for details.'
-exit 1
+cargo semver-checks --workspace --baseline-rev "$base_sha"
```

Comment:
One important distinction between the old script and the new script is that this will only ensure we bump the version if the API has changed.  Changes in the implementation will not be caught

Reply:
That's fair. Sounds like we inevitably requires a xtask/script to detect changes in member crates. Maybe we can keep lines 23-33 and extend them a bit?

(I am thinking just checking if any `.rs` `Cargo.toml` have changed)

Reply:
Just rewrote it in Rust. Should be a bit more friendly :)